### PR TITLE
Changes how detomatix pda detonation works

### LINKED ISSUE #4657: Detomatix cartridge not working properly
When used, and sucessfully, it does not blow up the target's PDA, despite the success admin log entry.
However, if it fails and backfires, it blows up your own PDA.

### PR DESCRIPTION
**Pull request by Alberyk**
-rewrites a bit of the code, to make it less terrible
-fixes #4657 by making sure that the pda properly explodes now
-removes relative pathing from the pda box

### CODE REVIEW

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -903,65 +903,56 @@ var/global/list/obj/item/device/pda/PDAs = list()
...
+				M.apply_effect(5, WEAKEN)
+				flick("flash", M.flash)
+			message += "Your [P] flashes with a blinding white light!"
+
+		if("spark")
+			spark(P.loc, 2)
+			message += "Your [P] begins to spark violently!"
+
+
+	if(message)
```
Comment (skull132) [inline code/game/objects/items/devices/PDA/PDA.dm]: This doesn't make sense. A possible outcome is this:

> Your PDA suddenly explodes! It melts in a puddle of plastic.

The way this was done before was that a random chance was utilized to determine how the PDA died. But this kinda railroads it into always melting, despite it also potentially exploding.

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 2]
```diff
@@ -903,65 +903,56 @@ var/global/list/obj/item/device/pda/PDAs = list()
...
+			message += "Your [P] suddenly explodes!"
+
+		if("burn")
+			if(M && isliving(M))
+				M.apply_damage( rand(30,60) , BURN)
+			message += "You feel a searing heat! Your [P] is burning!"
+
+		if("emp")
+			empulse(P.loc, 3, 6, 1)
+			message += "Your [P] emits a wave of electromagnetic energy!"
```
Comment (LordFowl) [inline code/game/objects/items/devices/PDA/PDA.dm]: Typically EMP effects do not have an attack log message, and empulse should already handle playing the sound.

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 3]
```diff
@@ -903,65 +903,56 @@ var/global/list/obj/item/device/pda/PDAs = list()
...
+			playsound(src, 'sound/effects/EMPulse.ogg', 50, 1)
+
+		if("smoke")
+			var/datum/effect/effect/system/smoke_spread/bad/S = new /datum/effect/effect/system/smoke_spread/bad
```
Comment (LordFowl) [inline code/game/objects/items/devices/PDA/PDA.dm]: I am pretty sure that S.start() already produces a sound, although I could be mistaken.

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 2 above)
Comment (Alberyk) [reply code/game/objects/items/devices/PDA/PDA.dm]: Removed the soud and message.

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 3 above)
Comment (Alberyk) [reply code/game/objects/items/devices/PDA/PDA.dm]: No, it does not.

**Pull request merged**